src/main/java/com/urbanairship/datacube/DataCubeIo.java

```diff
@@ -144,6 +162,16 @@ public Integer getValue() {
     public Optional<Future<?>> writeAsync(Batch<T> newBatch) throws AsyncException, InterruptedException {
         Batch<T> batchToFlush = null;
 
+        if (perRollupMetrics) {
+            newBatch.getMap().keySet().forEach(addr -> {
+                if (addr.getSourceRollup().isPresent()) {
+                    updateRollupCounter(rollupWriteIndividual, addr.getSourceRollup().get(), "rollupIndividualWrite");
+                } else {
+                    writeAddrWithoutRollup.inc();
```

Comment:
How would we hit this branch?

Reply:
You shouldn't be able to, but unlike in the read case where that decoration is performed locally, there's enough machinery between Address creation and this code that I'm not confident that I'm covering all current paths or possible future paths.